### PR TITLE
Small readme format updates

### PR DESCRIPTION
**Pull request by cartersocha**
Removed some small sections & added structure for the local quickstart.

### CODE REVIEW

**Comment by cartersocha** [comment]
Please review & merge @julianocosta89

File: README.md [hunk 1]
```diff
@@ -2,37 +2,44 @@
...
-## Quickstart
+## Local Quickstart
 
 ### Pre-requisites
 
 - Docker
 - [Docker Compose](https://docs.docker.com/compose/install/#install-compose) v2.0.0+
 
-### Steps
+### Step 1
```
Comment (reyang) [inline README.md:15]: minor: for folks who want to save the links/anchors, this might introduce problems. Doesn't have to block this PR though.

Comment (cartersocha) [reply README.md:15]: How would this potentially break links or introduce issues? I can adjust to a better layout

Comment (reyang) [reply README.md:15]: e.g. if something stored the current link, it'll be something like <https://github.com/cartersocha/opentelemetry-demo-webstore/tree/carterdocumentation2#step-3>, later if we added two more steps, it'll become `step-5`.

Comment (cartersocha) [reply README.md:15]: Ahh makes sense. I’ll adjust to actual step titles rather an ordered list

**Review by julianocosta89 — APPROVED** [review]

**Pull request merged**